resources/jenkins/release.sh

```diff
@@ -3,43 +3,37 @@ set -e
 
 # This script should be run after the build/tests have successfully completed
 
-MAJOR_MINOR=$1
-
 # Prune any local tags that aren't on the remote
 git fetch --prune origin "+refs/tags/*:refs/tags/*"
 
-DESCRIBE=`git describe --tags --always`
-REV=$(git log --pretty=format:'%h' -n 1)
+cd $WORKSPACE
 
-CURR_VERSION=`echo $DESCRIBE | awk '{split($0,a,"-"); print a[1]}'`
-echo "Got current git version $CURR_VERSION"
-MAJOR_VERSION=`echo $CURR_VERSION | awk '{split($0,a,"."); print a[1]}'`
-MINOR_VERSION=`echo $CURR_VERSION | awk '{split($0,a,"."); print a[2]}'`
+# Let's get version from maven
+MVNVER=`xmllint --xpath "/*[local-name()='project']/*[local-name()='version']/text()" pom.xml`
+TAG_NAME="v${MVNVER/-SNAPSHOT/}"
+echo $TAG_NAME
 
-if [ "$MAJOR_MINOR" == "Minor" ]; then
-    MINOR_VERSION=$((MINOR_VERSION+1))
-elif [ "$MAJOR_MINOR" == "Major" ]; then
-    MAJOR_VERSION=$((MAJOR_VERSION+1))
-    MINOR_VERSION=0
+if ! git rev-parse $TAG_NAME >/dev/null 2>&1
+then
+    git tag -a $TAG_NAME -m "Tagged automatically by Jenkins"
+	  git push origin $TAG_NAME
 else
-    echo "Error, unrecognized release type $MAJOR_MINOR.  Options are 'Major' and 'Minor'"
-    exit 1
+	echo "Tag: $TAG_NAME already exists."
```

Comment:
shouldn't this abort the build and exit with error?

Reply:
Nope, so now there is no tag 8.0. The first time it sees it creates it and this will be ver 8.0-0. This means major.minor 8.0 and 0 commits after major.minor change. Let's say we rebase a PR with 3 commits and next version will be 8.0-3, 3 commits after major.minor change.
Once we want to bump minor, let's say, we do that in pom.xml and the script will create 8.1 tag and describe git command will continue counting commits for us and the deb package version will be 8.1-0 for the first 8.1 build and so on and so on.